### PR TITLE
DO NOT MERGE:  demonstrate MKL failure

### PR DESCRIPTION
**Pull request by grlee77**
Here I have commented out the static linking proposed in #5 to demonstrate the MKL error

### CODE REVIEW

**Comment by jakirkham** [comment]
It appears this is passing. What are we missing here? 😕

**Comment by grlee77** [comment]
not sure.  It does seem to have installed mkl during the test phase

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,6 +8,8 @@ source:
     fn: pyFFTW-{{ version }}.tar.gz
     url: https://pypi.python.org/packages/source/p/pyFFTW/pyFFTW-{{ version }}.tar.gz
     md5: dab57de9d23f3b333115265914f3d240
+    patches:
+        - setup.patch
```
Comment (jakirkham) [inline recipe/meta.yaml]: Should we comment this out too?

**Comment by grlee77** [comment]
unfortunately I am not at work where my linux machine is at the moment so cannot try to compare with the environment there.

File: recipe/meta.yaml (hunk 1 above)
Comment (grlee77) [inline recipe/meta.yaml]: can do it, but it shouldn't matter.

**Comment by jakirkham** [comment]
Do you have docker-engine?

**Comment by grlee77** [comment]
trying with MKL 11.3.1.  I think that is the version I had been using when I was getting the failures.

**Comment by jakirkham** [comment]
Do you have [docker-machine]( https://docs.docker.com/machine/install-machine/ )? Maybe you can pull down an Ubuntu image and try debugging with it.

**Comment by grlee77** [comment]
I can install it and see.  Is there a particular image you would use?

**Comment by jakirkham** [comment]
So, the one we use at conda-forge is [condaforge/linux-anvil]( https://hub.docker.com/r/condaforge/linux-anvil/ ).

**Comment by jakirkham** [comment]
That's CentOS though. I don't know if there was something special about the Ubuntu version you were using previously.

**Comment by grlee77** [comment]
I installed the linux-anvil container and then started an interactive session via:

docker run -i -a stind -a stdout -a stderr condaforge/linux-anvil bash

Then the commands I ran were
```
conda install numpy -y
conda install -y -c conda-forge pyfftw
python -c "import numpy as np;  from pyfftw.interfaces.numpy_fft import fftn;  r = np.random.randn(32, 32, 32); F=fftn(r, axes=(0,))"
```
The python call above gives the following error in the container which is exactly what I had been seeing previously.  Any idea why this same error isn't occurring on the CI?
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/conda/lib/python3.5/site-packages/pyfftw/interfaces/numpy_fft.py", line 146, in fftn
    calling_func)
  File "/opt/conda/lib/python3.5/site-packages/pyfftw/interfaces/_utils.py", line 102, in _Xfftn
    FFTW_object = getattr(builders, calling_func)(*args)
  File "/opt/conda/lib/python3.5/site-packages/pyfftw/builders/builders.py", line 364, in fftn
    avoid_copy, inverse, real)
  File "/opt/conda/lib/python3.5/site-packages/pyfftw/builders/_utils.py", line 190, in _Xfftn
    flags, threads)
  File "pyfftw/pyfftw.pyx", line 1110, in pyfftw.pyfftw.FFTW.__cinit__ (/feedstock_root/build_artefacts/work/pyFFTW-0.10.1/pyfftw/pyfftw.c:10558)
RuntimeError: The data has an uncaught error that led to the planner returning NULL. This is a bug.
```

**Comment by jakirkham** [comment]
So, I tried to pin down the test dependencies in the recipe to see if I could repeat your results, but seem unable to reproduce them. However, this might be a nice starting point for determining what is going on. See this branch ( https://github.com/jakirkham/pyfftw-feedstock/tree/static_link_2 ). I have built with each commit and had no luck.

**Comment by grlee77** [comment]
You do not get the error if you run the commands above in the docker image?

**Comment by grlee77** [comment]
@jakirkham:  
Finally!  I suddenly realized what the problem was:  One of the workarounds to the MKL error is to make sure `pyfftw` is imported **prior** to `numpy`.  I purposefully imported `numpy` first in the test script so the failure would occur, but the additional import tests in `meta.yaml` were undermining the intended behavior!  I removed those tests from meta.yaml and now the failure occurs as expected!

**Comment by jakirkham** [comment]
Ah! That indeed makes sense. Glad you figured it out. 😄

Given this change, can we update PR ( https://github.com/conda-forge/pyfftw-feedstock/pull/5 ) respectively?

**Comment by grlee77** [comment]
yep.  just did

**Comment by grlee77** [comment]
closing this now that the failures can reliably be replicated on CI

**Pull request closed without merging**